build-now.lua

```diff
@@ -435,9 +449,21 @@ local function build_construction(bld)
     local construction_type = bld.type
     dfhack.buildings.deconstruct(bld)
 
-    -- add entries to df.global.world.constructions and adjust tiletypes for
-    -- the construction itself
-    create_and_link_construction(pos, item, false)
+    -- check if we're building on a construction (i.e. building a construction on top of a wall)
+    local tiletype = dfhack.maps.getTileType(pos)
+    local tileattrs = df.tiletype.attrs[tiletype]
+    if tileattrs.material == df.tiletype_material.CONSTRUCTION then
+        -- modify the construction to the new type
+        local construction, found = utils.binsearch(df.global.world.constructions, pos, "pos", pos_cmp)
+        if not found then
+            error("Error: could not find construction entry for construction tile at " .. pos.x .. ", " .. pos.y .. ", " .. pos.z)
+        end
+        reuse_construction(construction, item)
+    else
+        -- add entries to df.global.world.constructions and adjust tiletypes for
+        -- the construction itself
```

Comment:
the next line doesn't actually set the tiletype. is this an old comment?

Reply:
Yep oops